### DISCUSSION TITLE
Router region replacement funky with Query-No-Results block after Post-Template block

### DISCUSSION THREAD

**Discussion by poof86** [WordPress/gutenberg#60313] [Interactivity API]
It seems that the Query region isn't properly replacing all elements when a Query-No-Results block in present in the query region. 

For example, I have a spacer block after the Post-Template block that is not included in the region update. While the following blocks after this spacer are included. I've checked the html that the browsers receives in the Dev Console (under Network) and the spacer block is there in the html.

When I place the core Query-No-Results block near the Post-Template block, I notice that the Post Template block doesn't get emptied/removed all the way when navigating to a query with zero results.

The Query-No-Results block will be added above the Post-Template block which only holds one post at this point. The mess keeps growing when you navigate to a query with results and the previous Post-Template block isn't replaced. Navigating to empty results again, another Query-No-Results is added beneath the already rendered Query-No-Results block etc.

So it's version 1 where something extra is removed, or version 2 where blocks aren't properly removed.

I wasn't sure if I should make an issue for this, or to share it here in the discussion.

**Comment by poof86** [comment]
Sidenote: I think attributes without a "=" also aren't loaded in correctly. for instance the attribute 'disabled' on it self is ignored during region replacement? This is without any wp-bind directive

**Comment by luisherranz** [comment]
How do you navigate to a Query with no results if it doesn't have pages, and therefore no navigation links?

**Reply by poof86** [reply]
I'm not sure what you mean by navigating to a query that doesn't have pages and navigation links.

The search template page contains the Core Query Block with a Post Template. The Query Block is setup to use the main query which is influenced by the query vars in the url the user navigates to. Sometimes this results in a query with no results.

My client didn't want to display any query results before a search term is entered or when the result is empty. So the main search query is short-circuited to not display all posts when the query did not find any posts.

Everything works fine when I leave out the Query-No-Results Block.

**Reply by luisherranz** [reply]
Oh, I see. Sorry, I thought your issue was with the Query block, not with a block you're creating yourself.

Yes, right now, region-based client-side navigation cannot be used if there are different blocks in each navigation. But we are working on it and we will try to solve this problem in a future version.

As a workaround, you can include a hidden No Results block in the rendered HTML even when there are results, so that the CSS of the No Results block is always loaded.
